recipes/boost/all/test_package/conanfile.py

```diff
@@ -1,21 +1,50 @@
 from conans import ConanFile, CMake, tools
 import os
+import shutil
 import sys
 
 
 class DefaultNameConan(ConanFile):
     settings = "os", "compiler", "arch", "build_type"
-    generators = "cmake"
+    generators = "cmake", "cmake_find_package"
+
+    @property
+    def _conan_build_folder(self):
+        return "conan_find_package"
+
+    @property
+    def _cmake_build_dir(self):
+        return "cmake_find_package"
 
     def build(self):
+        # Move the generated FindBoost.cmake to backup/FindBoost.cmake
+        os.mkdir("backup")
+        os.rename(src=os.path.join("FindBoost.cmake"), dst=os.path.join("backup", "FindBoost.cmake"))
+
+        # Build using CMake's find_package
+        # FIXME: disabled because recent boost versions require a recent cmake versions.
```

Comment:
question: does setting `Boost_ADDITIONAL_VERSIONS` help to discover new boost version with older CMake releases?

Reply:
No, it will just not generate any imported targets.
See https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/FindBoost.cmake#L1187

The latest iteration of this pr uses the `cmake_find_package` conan generator in test_package.
The generated module works fine on all cmake versions.

Testing whether a cmake version supports boost might be possible, but would not be stable.
https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/FindBoost.cmake#L1457

Reply:
okay, that's unfortunate but expected. leave it as is.